From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Notify non-GPL module loading will be going away in January 2008

Numerous kernel developers feel that loading non-GPL drivers into the
kernel violates the license of the kernel and their copyright.  Because
of this, a one year notice for everyone to address any non-GPL
compatible modules has been set.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 Documentation/feature-removal-schedule.txt |    9 +++++++++
 kernel/module.c                            |    7 ++++++-
 2 files changed, 15 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/Documentation/feature-removal-schedule.txt
+++ gregkh-2.6/Documentation/feature-removal-schedule.txt
@@ -272,3 +272,12 @@ Why:	Speedstep-centrino driver with ACPI
 Who:	Venkatesh Pallipadi <venkatesh.pallipadi@intel.com>
 
 ---------------------------
+
+What:	non-GPL-licensed modules will no longer be loaded.
+When:	January 2008
+Why:	Numerous kernel developers feel that loading non-GPL drivers into the
+	kernel violates the license of the kernel and their copyright.
+
+Who:	Greg Kroah-Hartman <greg@kroah.com> <gregkh@novell.com>
+
+---------------------------
--- gregkh-2.6.orig/kernel/module.c
+++ gregkh-2.6/kernel/module.c
@@ -1393,9 +1393,14 @@ static void set_license(struct module *m
 		license = "unspecified";
 
 	if (!license_is_gpl_compatible(license)) {
-		if (!(tainted & TAINT_PROPRIETARY_MODULE))
+		if (!(tainted & TAINT_PROPRIETARY_MODULE)) {
 			printk(KERN_WARNING "%s: module license '%s' taints "
 				"kernel.\n", mod->name, license);
+			printk(KERN_WARNING "%s: This module will not be able "
+				"to be loaded in any kernel released after "
+				"January 1, 2008 due to its license.\n",
+				mod->name);
+		}
 		add_taint_module(mod, TAINT_PROPRIETARY_MODULE);
 	}
 }
